### PR TITLE
Add domain rendering script

### PR DESCRIPTION
**Pull request by nilsvu**
Shows elements and grid lines. A second high-resolution dataset can be used to draw smooth element outlines. This rendering is a starting point for visualizations of the domain geometry, e.g. for publications.

For example:

punctures_domain

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by yoonso0-0** [comment]
output

**Comment by yoonso0-0** [comment]
It's working really well. Thanks for implementing this.

The above output image is what I get when I fix the clip normal to `0.0, 0.0, 1.0` (Z axis, default) but shifted the clip origin into `10.0, 0.0, 0.0`. In principle it should show the same image as below : 

output


Is this a paraview behavior or a bug is somewhere?

**Comment by nilsvu** [comment]
Should be fixed (see fixup commit). Camera positioning is a bit tricky with ParaView... I'm sure this can be made more robust, but I think this is ok for a start.

**Comment by yoonso0-0** [comment]
Checked the fixup solved issue. Please squash

File: src/Visualization/Python/Render3D/Domain.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    pv.ColorBy(outline_display, None)
+
+    # Set resolution
+    layout = pv.GetLayout()
+    layout.SetSize(1200, 1200)
+
+    # Configure camera
+    camera = pv.GetActiveCamera()
+    camera.SetFocalPoint(*clip_origin)
+    camera.SetPosition(*(np.array(clip_origin) + np.array(clip_normal) * 10.0))
```
Comment (knelli2) [inline src/Visualization/Python/Render3D/Domain.py:178]: This factor of 10 seems a bit arbitrary. I could imagine this not working for a BBH with an outer boundary of like 1000.

**Review by knelli2 — APPROVED** [review]
Let me know if you want to do anything about my suggestion. Not necessary though

File: src/Visualization/Python/Render3D/Domain.py (hunk 1 above)
Comment (nilsvu) [reply src/Visualization/Python/Render3D/Domain.py:178]: Actually ParaView ignores this factor, so I kept it at 10 :D It sets the zoom factor automatically to show the entire domain. As I said above, we can surely make the camera handling better here, but I haven't played around with it enough to understand what ParaView actually does with the camera.

**Pull request merged**